desktop-widgets/tab-widgets/TabDiveInformation.cpp

```diff
@@ -247,7 +247,12 @@ void TabDiveInformation::updateTextBox(int event) // Either the text box has bee
 				ui->atmPressVal->setText(ui->atmPressVal->text().sprintf("%d",atmpress.mbar));
 				ui->atmPressType->setCurrentIndex(0);    // reset combobox to mbar
 			} else { // i.e. event == COMBO_CHANGED, that is, "m" or "ft" was selected from combobox
-				ui->atmPressVal->clear();   // Clear the text box so that altitude can be typed
+				 // Show estimated altitude
+				bool ok;
+				int i = ui->atmPressVal->text().toInt(&ok,10);
+				if (ok && ui->atmPressVal->text().length()) {
+					ui->atmPressVal->setText(QString::number(pressure_to_altitude(i)/1000));
+				}
```

Comment:
so the user changed the label to 'm' or 'ft', and then you convert the pressure to an altitude, so `i` should have been named `pressureValue` or somethings like that. But that's where I'm confused, because if the user changed this to 'm' or 'ft', don't we assume that the input value is in meters or feet?

Or is the logic that there is a pressure value in the field, the user switches to 'm', we convert that value into an estimated altitude, and THEN the user edits the value? I guess that would make more sense, but then a comment here in the code would be helpful (`// convert existing pressure value into estimated altitude`)

Reply:
I discovered I did not provide for the imperial case. Here it is.
The logic is as follows:
1) Dive has no surface pressure value. However, diver knows altitude (e.g. 3000 ft)
2) Select ft from the combo box and type in the value of 3000 in the text box. Hit ENTER or click mouse elsewhere.
3) The estimated atm pressure for 3000 ft is calculated and shown (900 mbar)
4) Now let us assume the dive had an atm pressure of 900 mbar (maybe even from elsewhere e.g. dive computer)
    We are curious to know to what altitude this relates to.
5) The text box shows 900 and the combobox shows mbar.
6) Select ft in combobox. The estimated altitude in ft is now shown (some value near 3000 ft)
7) Selecting mbar from the combobox shows 900 mabr again, as previously.

Reply:
yeah, the explanation makes sense. Obviously this conversion is a "rough guess", but I can see how some people might appreciate it.
See my other comment about the include lines added (I think) by mistake...